lib/chef/resource/registry_key.rb

```diff
@@ -75,12 +77,12 @@ def key(arg = nil)
         )
       end
 
-      def values(arg = nil)
-        if not arg.nil?
-          if arg.is_a?(Hash)
-            @values = [ arg ]
-          elsif arg.is_a?(Array)
-            @values = arg
+      property :values, [ Hash, Array ], default: nil, sensitive: true, desired_state: false, coerce: proc { |v|
+        if not v.nil?
+          if v.is_a?(Hash)
+            @values = [ v ]
```

Comment:
The coerce method should be returning the modified value, rather than directly touching instance variables.